tools/reviewer/job/sync/GitHubClient.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2018 The StartupOS Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.startupos.tools.reviewer.job.sync;
+
+import org.json.JSONObject;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Base64;
+
+import static java.net.HttpURLConnection.HTTP_CREATED;
+import static java.net.HttpURLConnection.HTTP_OK;
+
+public class GitHubClient {
```

Comment:
I suggest this - 
1. GitHubClient will be "smarter" and will contain all the calls to GitHub - urls, requests, responses.
2. Every request and response will have a proto. Instead of working with JSONObject (which is not nice-looking) and the hard-coded strings, we will work with protos directly, and automatically convert the json string to proto.
3. GitHub client will receive request protos and return response protos. `postData` and `getResponse` will become private.
4. The remaining code in GitHubReader and GitHubWriter will be responsible of handling the request and response protos and converting them to what we actually need in GitHubSync.

This way, the client is responsible for all GitHub "implementation details" except for the API, which becomes a clean proto API.